Assets/Code/data classes/visibilty.cs

```diff
@@ -3,112 +3,189 @@
 using UnityEngine;
 using System;
 
-public class visibility
+public static class visibility
 {
+  [Flags]
+  public enum raycastParameters
+  {
+      planet = 1,
+      satellite = 2,
+      facility = 4,
+      all = 8
+  }
 
-  public bool raycast(position position1, position position2, int flags, int defaultRadius = 1)
+  public struct raycastInfo
+  {
+
+    public bool hit;
+    public List<planet> hitPlanets;
+    public List<facility> hitFacs;
+    public List<satellite> hitSats;
+
+    public raycastInfo(bool hit, List<planet> HitPlanets, List<satellite> HitSats, List<facility> HitFacs)
+    {
+      this.hit = hit;
+      this.hitPlanets = HitPlanets;
+      this.hitSats = HitSats;
+      this.hitFacs = HitFacs;
+    }
+  }
+
+  public static double distance(position p1, position p2)
+  {
+     double distance = Math.Sqrt(Math.Pow((p1.x - p2.x) ,2) + Math.Pow((p1.y - p2.y) ,2) + Math.Pow((p1.z - p2.z), 2));
+     return distance;
+  }
+
+  public static double raycastMath(position position1, position position2, position obj)
+  {
+    double distance1 = distance(position1, obj);
+    double distancePercent = distance1 / distance(position1, position2);
+
+    position pointOnLine = new position((position1.x + position2.x) * distancePercent,
+    (position1.y + position2.y) * distancePercent,
+    (position1.z + position2.z) * distancePercent);
+
+    double distanceFromObj = distance(pointOnLine, obj);
+
+    return distanceFromObj;
+
+  }
+
+  //double.max
```

Comment:
Before you submit code, please clean it up. This means removing unnecessary code and empty lines.